cargo-miri/src/phases.rs

```diff
@@ -541,6 +541,27 @@ pub fn phase_runner(mut binary_args: impl Iterator<Item = String>, phase: Runner
     }
 }
 
+fn get_miriflags() -> Option<Vec<String>> {
+    // Fetch miri flags from cargo config.
+    let mut cmd = cargo();
```

Comment:
I think we need to do a bit more work here: the user might have passed `--config ...` to `cargo miri run`, which would be ignored this way. See `get_cargo_metadata` for how to forward that. Also they might have set the manifest path, which even `get_cargo_metadata` gets wrong.

Reply:
Once https://github.com/rust-lang/miri/pull/2474 lands there will be a function `cargo_extra_flags` you can call to get all the flags you need.

However, you can only call that function in `phase_cargo_miri`; the arguments are lost here in `phase_runner`. So you'll need to do the flag handling there, and then put the results into an env var that is picked up again in `phase_runner`.

Reply:
I think the "right" way to do that is to have a variable like CARGO_ENCODED_MIRIFLAGS. Then phase_cargo_miri can determine the flags (from either MIRIFLAGS or via the config -- or from CARGO_ENCODED_MIRIFLAGS if that is already set -- and then store the flags losslessly (without spaces causing trouble) into CARGO_ENCODED_MIRIFLAGS here phase_runner can retreive it.

If that's more work than what you signed up for, then storing it into MIRIFLAGS and not supporting spaces within arguments for now would also work.